lib/6to5/generation/whitespace.js

```diff
@@ -11,15 +11,17 @@ Whitespace.prototype.getNewlinesBefore = function (node) {
   var startToken;
   var endToken;
   var tokens = this.tokens;
+  var token;
+
+  for (var i = 0; i < tokens.length; i++) {
```

Comment:
and as you can see [here](https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#32-leaking-arguments) bluebird also caches array lengths